Source/CronosPlayUnreal/Private/PlayCppSdkActor.cpp

```diff
@@ -93,6 +98,38 @@ void APlayCppSdkActor::Destroyed() {
   assert(NULL == _sdk);
 }
 
+void APlayCppSdkActor::ConnectWalletConnect(FString description, FString url,
+                                            TArray<FString> icon_urls,
+                                            FString name, int64 chain_id) {
+  FString Jsondata;
+  bool IsRetored;
+  FString RestoreClientOutputMessage;
+  RestoreClient(Jsondata, IsRetored, RestoreClientOutputMessage);
+  if (IsRetored) {
+    // Setup Callback
+    bool IsSetupCallback;
+    FString SetupCallbackOutputMessage;
+    OnReceiveWalletconnectSessionInfoDelegate.BindDynamic(
+        this, &APlayCppSdkActor::OnWalletconnectSessionInfo);
+    SetupCallback(OnReceiveWalletconnectSessionInfoDelegate, IsSetupCallback,
+                  SetupCallbackOutputMessage);
+    if (IsSetupCallback) {
+      // Ensure session
+      OnEnsureSessionDelegate.BindDynamic(this,
+                                          &APlayCppSdkActor::OnRestoreSession);
+      EnsureSession(OnEnsureSessionDelegate);
+    } else {
+      // TODO
```

Comment:
would there be a situation where one wouldn't set up a callback? should this return or log an error?

Reply:
Done